From 504f017876acb1a2514d28f955a685c3d7d9bfd8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Erdmann?= <dywi@mailerd.de>
Date: Mon, 24 Mar 2014 20:52:10 +0100
Subject: [PATCH 1/3] systemd-compat

---
 49tlp       | 4 ++++
 tlp-rf-func | 8 ++++++++
 2 files changed, 12 insertions(+)

diff --git a/49tlp b/49tlp
index 20ff103..8b2367f 100755
--- a/49tlp
+++ b/49tlp
@@ -6,6 +6,10 @@
 
 . "${PM_FUNCTIONS}"
 
+# *** systemd-compat ***
+#  do nothing if systemd is active
+[ ! -d /run/systemd/system ] || exit $NA
+
 case $1 in
 	hibernate|suspend)
 		tlp suspend
diff --git a/tlp-rf-func b/tlp-rf-func
index e458dbc..907bf3a 100755
--- a/tlp-rf-func
+++ b/tlp-rf-func
@@ -417,6 +417,14 @@ init_radio_devices () { # initialize all radio type states
 
                 # NM workaround: when not explicitly configured,
                 # re-enable wifi on shutdown to prepare for startup
+
+                # *** systemd-compat ***
+                #  Disabling this on systemd-booted machines as it
+                #  might interfere with systemd-rfkill@.service,
+                #  introduced in systemd 209.
+                #   http://www.freedesktop.org/software/systemd/man/systemd-rfkill@.service.html
+                #
+                [ -d /run/systemd/system ] || \
                 if ! wordinlist "wifi" "$devs2disable $devs2enable"; then
                     devs2enable="wifi $devs2enable"
                 fi
-- 
1.9.3

